zebedee-cms/src/main/java/com/github/onsdigital/zebedee/keyring/KeyringImpl.java

```diff
@@ -102,12 +103,41 @@ public SecretKey get(User user, Collection collection) throws KeyringException {
 
     @Override
     public void remove(User user, Collection collection) throws KeyringException {
-        // TODO
+        validGetParams(user, collection);
+
+        boolean hasAccess = false;
+        try {
+            hasAccess = permissionsService.canEdit(user, collection.getDescription());
+        } catch (IOException ex) {
+            throw new KeyringException(ex);
+        }
+
+        if (!hasAccess) {
+            return;
+        }
+
+        cache.remove(collection.getDescription().getId());
     }
 
     @Override
     public void add(User user, Collection collection, SecretKey key) throws KeyringException {
-        // TODO
+        validGetParams(user, collection);
+        if (key == null) {
+            throw new KeyringException(SECRET_KEY_NULL_ERR);
+        }
```

Comment:
May as well create a `validateKey(SecretKey key)` method too. The logic is very small/straight forward but I think calling a method `validateKey(key)` reads more like english and is much more descriptive and clear to anyone looking at this for the first trime.